### PR TITLE
add dashboard.nvim, neogit and emmet

### PR DESCRIPTION
**Pull request by imMaturana**
Sorry about the lack of description in neogit module, unfortunately the neogit doc isn't clear

Also, sorry about my English

### CODE REVIEW

**Comment by pta2002** [comment]
I'll check out the docs you mentioned to see if I can add anything, otherwise I think it's good!

**Comment by imMaturana** [comment]
After looking through the [README](https://github.com/TimUntersberger/neogit/blob/master/README.md) I saw how stupid I am, the descriptions are in comments format in the [configuration](https://github.com/TimUntersberger/neogit#configuration), sorry about this, I've added the descriptions

File: plugins/git/neogit.nix [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      signs = mkOption {
+        description = "Customize displayed signs";
+        type = types.submodule {
+          options = {
+            section = mkOption {
+              type = types.nullOr (types.listOf types.str);
```
Comment (pta2002) [inline plugins/git/neogit.nix:87]: Could you add a description for these?

File: plugins/git/neogit.nix [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+          };
+        };
+        default = { };
+      };
+
+      sections = mkOption {
+        description = "Section configuration";
+        type = types.submodule {
+          options = {
+            untracked = mkOption {
```
Comment (pta2002) [inline plugins/git/neogit.nix:123]: And here

File: plugins/utils/dashboard.nix [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+          };
+        };
+        default = { };
+      };
+
+      fzf = mkOption {
+        description = "Some options for fzf";
+        type = types.submodule {
+          options = {
+            float = mkOption {
```
Comment (pta2002) [inline plugins/utils/dashboard.nix:107]: Here too

**Comment by imMaturana** [comment]
I've added the submodules descriptions.

The `programs.nixvim.plugins.dashboard.fzf.float` option isn't working, I'll open a issue on [dashboard.nvim](https://github.com/glepnir/dashboard-nvim) about this.

**Pull request merged**